corehq/ex-submodules/casexml/apps/case/util.py

```diff
@@ -240,3 +239,35 @@ def get_all_changes_to_case_property(case, case_property_name):
             case_property_changes.append(property_changed_info)
 
     return case_property_changes
+
+
+def get_paged_changes_to_case_property(case, case_property_name, start=0, per_page=50):
+    """Return paged changes to case properties, and last transaction index checked
+    """
+
+    def iter_transactions(transactions):
+        for i, transaction in enumerate(transactions):
+            property_changed_info = property_changed_in_action(transaction, case.case_id, case_property_name)
+            if property_changed_info:
+                yield property_changed_info, i + start
+        raise StopIteration
+
+    num_actions = len(case.actions)
+    if start > num_actions:
+        return [], -1
+
+    case_transactions = iter_transactions(
+        sorted(case.actions, key=lambda t: t.server_date, reverse=True)[start:]
```

Comment:
this slice doesn't guarantee that there won't be overlap in the returned data

Reply:
Perhaps I'm not understanding what `start` is but I think its the value from the UI which represents how far you are through the pagination. In this case the UI doesn't know how many transactions you went through to generate the page so if you went through 20 transactions to get a page of 10 and the `start` param came back as 11 for the start of the next page then we could get some duplicate changes as we're starting at transaction 11 instead of 21.

Hope that makes sense and feel free to ignore if I've got it wrong.

Reply:
`start` actually returns the next _transaction_ that needs to be checked, not the number of items that has been shown on the screen. So if it goes through 100 transactions to only show 50 items, the `start` value will be 101. This might be confusing, although I did name the frontend variables with what I thought to be pretty clear names (`lastTransactionChecked`, `nextTransaction`, etc) . Let me know if I could have made this clearer in the code somehow.